### PR TITLE
Feature: Adding audit structure to product repository and test

### PR DESCRIPTION
**Pull request by eapg**
* adding audit structure to product repository

### CODE REVIEW

File: src/constants/audit.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from enum import Enum
+
+
+class AuditStatus(Enum):
```
Comment (reynoldmorel) [inline src/constants/audit.py:4]: I think you can simplify:
```
enum Status {
  ACTIVE = "ACTIVE"
  DELETED = "DELETED"
}

enum INTERNAL_USERS {
   SEEDER = -1
  KITCHEN_SIMULATOR = -2 
}
```

File: src/lib/entities/product.py [hunk 2]
```diff
@@ -10,9 +10,11 @@ def __init__(self):
         self.description = None  # string
         self.entity_status = None  # enum
         self.create_date = None  # date
+        self.delete_date = None  # date
```
Comment (reynoldmorel) [inline src/lib/entities/product.py:13]: Could you create `AbstractEntity` that would contain:
1. id
2. entity_status
3. created_date
5. created_by
6. updated_date
7. updated_by

Finally make this class to be inherited by product and other entities

File: src/lib/repositories/generic_repository.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ def get_all(self):
         pass
 
     @abstractmethod
-    def delete_by_id(self, obj_id):
+    def delete_by_id(self, obj_id, product):
```
Comment (reynoldmorel) [inline src/lib/repositories/generic_repository.py:20]: maybe change `product` by `obj`?

File: src/lib/repositories/impl/product_repository_impl.py [hunk 4]
```diff
@@ -10,19 +12,39 @@ def __init__(self):
 
     def add(self, product):
         product.id = self._current_id
+        product.create_date = datetime.now()
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/product_repository_impl.py:15]: where is `product.udpated_by = product.created_by` same comment for the dates?

File: src/lib/repositories/impl/product_repository_impl.py [hunk 5]
```diff
@@ -10,19 +12,39 @@ def __init__(self):
...
+        products = list(self._products.values())
+        products_filtered = list(
+            filter(
+                lambda product: product.entity_status == AuditStatus.ACTIVE, products
+            )
+        )
+        return list(products_filtered)
+
+    def delete_by_id(self, product_id, product):
+        product.delete_date = datetime.now()
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/product_repository_impl.py:39]: did you mean `updated_date`?

Also can you do a `get_by_id` here? the only thing you need right now from `product` is the `updated_by`.

Should look like:

```
product_to_delete = self.get_by_id(product_id)
product_to_delete.entity_status = DELETED
product_to_delete.updated_date = now()
product_to_delete.updated_by = product.updated_by
self.update_by_id(product_id, product_to_delete)
```

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/constants/audit.py (hunk 1 above)
Comment (eapg) [reply src/constants/audit.py:4]: done

File: src/lib/entities/product.py (hunk 2 above)
Comment (eapg) [reply src/lib/entities/product.py:13]: done.

File: src/lib/repositories/generic_repository.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories/generic_repository.py:20]: done.

File: src/lib/repositories/impl/product_repository_impl.py (hunk 5 above)
Comment (eapg) [reply src/lib/repositories/impl/product_repository_impl.py:39]: done.

File: src/lib/repositories/impl/product_repository_impl.py (hunk 4 above)
Comment (eapg) [reply src/lib/repositories/impl/product_repository_impl.py:15]: done.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**